web-console/knikubevirt/vm-details/vm-details.md

```diff
@@ -10,6 +10,33 @@ The user sees detailed information about the VM’s configuration.
 ![VM-overview - Editing a field](img/1-0-1.jpg)
 Editing field will be aligned to the rest of the system. For now, only 'Description' and Flavor' have a pencil icon next to the field name to indicate it is editable. Hitting that pencil will open up a modal.
 
+### Editing the CD
+Via the pencil icon the cd can be edited.
+![VM-details edit cd](img/cd-edit.png)
+
+They will then be presented a modal with a type to select from.
+
+Depending on the type selected a new input is revealed. These types would be
+#### No CD
+This sets CD to none.
+![VM-details edit cd modal](img/edit-blank.png)
+#### Container
+A text field is provided to paste in the path to the container.
+![container-type](img/edit-container.png)
+#### Persistent Volume Claim
+A typehead field is provided to find the desired Persistent volume claim. 
+![PVC-type](img/edit-pvc.png)
+#### URL
+A text field is provided to paste in the URL to the available CD.
+![URL-type](img/edit-url.png)
+
+With each selection a tooltip is provided to describe each type.
+
+### Windows drivers
+In the case where a Windows operating system has been chosen in the general step a checkbox is provided to confirm this cd will have the windows drivers attached to it.
+![VM-details edit cd windows drivers](img/edit-container-windows.png)
```

Comment:
Sorry for so many questions, just learning. 😄

Does this mean that the virtual CD/DVD/disk will *contain* Windows drivers? Attaching drivers to a virtual CD that'll then be attached to the VM sounds a bit odd. Maybe helper text below the checkbox could be helpful to explain what this checkbox is used for?

And is this driver CD necessary after the drivers are successfully installed? Would we be able to handle removing this virtual CD automatically? Would the user have any use for this checkbox after they're installed?

Reply:
good questions! I'm still trying to wrap my head around exactly how this works. I put this design in here because I want @jelkosz have a look and give some feedback on it. 
That's the reason I have the (?) there too, I think it might be necessary if the checkbox label doesn't make it super clear.

Reply:
There is a CD containing the drivers available (as a container disk). So in theory you should be able to survive without the checkbox and let the user to figure out where this drivers are. The problem is that this would mean that the user would have to read the documentation first :) 

The idea with the checkbox was that since we provide the drivers anyway and we know where they are and this is such a common case that a shortcut for the user makes sense. He will just check the checkbox and we will automatically pick the container disk and configure the correct path to it.

The problem with automatically removing the drivers is that we don't really know when the drivers have been installed. We know that the user attached the drivers, booted the VM and than turned it off so he probably spent his time in VM installing drivers and he has probably succeeded so we might remove the CD and he will probably appreciate this. I don't know, what you think?

Reply:
thanks @jelkosz that's very helpful. 
I think these designs need updating then. If we can provide the container then there's no need to select one. Could we rather, from the dropdown provide a `Windows drivers` option?

As far as removing after they have attached, booted, shut down - I wonder what the risk of removing them prematurely is. It would be very confusing if you forgot to install and then remember and went back and they were gone. 

What is the cost of leaving them and letting the user remove them manually?

Reply:
<img width="544" alt="Screen Shot 2019-08-21 at 11 57 27 AM" src="https://user-images.githubusercontent.com/287880/63462526-824c7080-c431-11e9-81f8-2911dbcd40a4.png">

Reply:
yes, this looks very good, given the "name-of-container-with-drivers" is generated automatically.

> What is the cost of leaving them and letting the user remove them manually?

Couple:
1: you will have the CD in the VM you don't use (e.g. just strange, nothing important)
2: after you turn the VM off, you will have to do couple of more clicks (and remember to do them) to remove the CD
3: probably the only actually problematic one: Every time the VM is started and the container disk is not cached on the host (e.g. no VM with the drivers have ever been started on that host), the container will be pulled again. Meaning, if you have 2 hosts in the cluster and create your only windows VM, start it with the container disk, it will be pulled and cached on host A. Than turn the VM off, forget to remove the CD even if you don't need it. Start the VM again, it will be scheduled on host B, the disk will be pulled and cached on host B even if you don't need it.

Reply:
hmm I wonder if we could address this in the status of the VM to make them aware that there is a CD That can be removed if it's no longer needed. "If you've already installed your drivers this CD can be removed". It could also be added to the VM details. WDYT @jelkosz ?